module/applications/components/item-list-controls.mjs

```diff
@@ -297,13 +322,14 @@ export default class ItemListControlsElement extends HTMLElement {
   _applySorting() {
     const comparators = {
       a: (a, b) => a.name.localeCompare(b.name, game.i18n.lang),
+      p: (a, b) => ((a.preparationMode === "always" ? 0 : 1)-(b.preparationMode === "always" ? 0 : 1)) || (a.prepared-b.prepared) || a.name.localeCompare(b.name, game.i18n.lang),
       m: (a, b) => a.sort - b.sort
     };
     for ( const section of this.list.querySelectorAll(".items-section .item-list") ) {
       const items = [];
       section.querySelectorAll(".item").forEach(element => {
-        const { itemName, itemSort } = element.dataset;
-        items.push({ element, name: itemName, sort: Number(itemSort) });
+        const { itemName, itemSort, itemPreparationMode, itemPreparationPrepared } = element.dataset;
+        items.push({ element, name: itemName, sort: Number(itemSort), preparationMode: itemPreparationMode, prepared: itemPreparationPrepared === "true" ? 0 : 1 });
```

Comment:
Might be over our configured line length limit. Also, it seems we could use the same technique of coercing the prepared value to a number to make our priority comparator simpler:

```suggestion
        items.push({
          element,
          name: itemName,
          sort: Number(itemSort),
          preparationMode: itemPreparationMode === "always" ? 0 : 1,
          prepared: itemPreparationPrepared === "true" ? 0 : 1
        });
```